Support TS flag resolveJsonModule

In TS 2.9, a compiler flag [resolveJsonModule](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-9.html#new---resolvejsonmodule) was added. Using this in my TS code, I get the following exception with karma-typescript:

Code:
```ts
import data from '../package.json';
```

Output:
```
<timestamp>:INFO [compiler.karma-typescript]: Compiling project using Typescript 3.0.3
<timestamp>:ERROR [compiler.karma-typescript]: src/test(1,8): error TS1192: Module '"<abs folder dir>/package"' has no default export.
```

Tried with latest pull from master. Getting a lot of type errors for jasmine/node, but also seeing errors like `Error: Cannot find module 'rgb-color'` during the bower test, and one of these:
```
> tests-integration-latest@3.0.13 dev:ci:core <repo-root>\tests\integration-latest
> gulp dev:ci --cwd . && karma start karma.conf.core.js --browsers ChromeHeadless --single-run

[14:21:52] Using gulpfile <repo-root>\gulpfile.js
[14:21:52] Starting 'clean'...
[14:21:52] Starting 'copy'...
[14:21:52] Finished 'clean' after 127 ms
[14:21:52] Finished 'copy' after 209 ms
[14:21:52] Starting 'dev:ci'...
[14:21:52] Finished 'dev:ci' after 53 μs
03 10 2018 14:21:56.240:INFO [compiler.karma-typescript]: Compiling project using Typescript 2.9.2
03 10 2018 14:22:01.056:INFO [compiler.karma-typescript]: Compiled 2 files in 4787 ms.
03 10 2018 14:22:02.577:INFO [bundler.karma-typescript]: Bundled imports for 2 file(s) in 1017 ms.
03 10 2018 14:22:02.661:INFO [karma]: Karma v2.0.5 server started at http://0.0.0.0:9876/
03 10 2018 14:22:02.665:INFO [launcher]: Launching browser ChromeHeadless with unlimited concurrency
03 10 2018 14:22:02.682:INFO [launcher]: Starting browser ChromeHeadless
03 10 2018 14:22:03.549:INFO [HeadlessChrome 0.0.0 (Windows 10 0.0.0)]: Connected on socket FoJlUBu3ghCR3C_pAAAA with id 97096932
HeadlessChrome 0.0.0 (Windows 10 0.0.0) ERROR
  {
    "message": "Uncaught Error: Can't find constants [<repo-root>/node_modules/constants-browserify/constants.json] (required by <repo-root>/tests/integration-latest/src/node/core/core-module-tester.ts)\nat <repo-root>/src/client/commonjs.js:13:17\n\nError: Can't find constants [<repo-root>/node_modules/constants-browserify/constants.json] (required by <repo-root>/tests/integration-latest/src/node/core/core-module-tester.ts)\n    at require (<repo-root>/src/client/commonjs.js:13:23)\n    at <repo-root>/src/client/commonjs.js:18:24\n    at Object.global.wrappers.<repo-root>/tests/integration-latest/src/node/core/core-module-tester.ts.assert (src/node/core/core-module-tester.ts:4 <- src/node/core/core-module-tester.js:26:29)\n    at require (<repo-root>/src/client/commonjs.js:17:25)\n    at <repo-root>/src/client/commonjs.js:18:24\n    at Object.global.wrappers.<repo-root>/tests/integration-latest/src/node/core/core-module-tester.spec.ts.events (src/node/core/core-module-tester.spec.ts:2 <- src/node/core/core-module-tester.spec.js:4:28)\n    at require (<repo-root>/src/client/commonjs.js:17:25)\n    at <repo-root>/src/client/commonjs.js:32:9\n    at Array.forEach (<anonymous>)\n    at <repo-root>/src/client/commonjs.js:31:40",
    "str": "Uncaught Error: Can't find constants [<repo-root>/node_modules/constants-browserify/constants.json] (required by <repo-root>/tests/integration-latest/src/node/core/core-module-tester.ts)\nat <repo-root>/src/client/commonjs.js:13:17\n\nError: Can't find constants [<repo-root>/node_modules/constants-browserify/constants.json] (required by <repo-root>/tests/integration-latest/src/node/core/core-module-tester.ts)\n    at require (<repo-root>/src/client/commonjs.js:13:23)\n    at <repo-root>/src/client/commonjs.js:18:24\n    at Object.global.wrappers.<repo-root>/tests/integration-latest/src/node/core/core-module-tester.ts.assert (src/node/core/core-module-tester.ts:4 <- src/node/core/core-module-tester.js:26:29)\n    at require (<repo-root>/src/client/commonjs.js:17:25)\n    at <repo-root>/src/client/commonjs.js:18:24\n    at Object.global.wrappers.<repo-root>/tests/integration-latest/src/node/core/core-module-tester.spec.ts.events (src/node/core/core-module-tester.spec.ts:2 <- src/node/core/core-module-tester.spec.js:4:28)\n    at require (<repo-root>/src/client/commonjs.js:17:25)\n    at <repo-root>/src/client/commonjs.js:32:9\n    at Array.forEach (<anonymous>)\n    at <repo-root>/src/client/commonjs.js:31:40"
  }
HeadlessChrome 0.0.0 (Windows 10 0.0.0): Executed 0 of 0 ERROR (0.261 secs / 0 secs)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
...
```

--- a/src/bundler/bundle-item.ts
+++ b/src/bundler/bundle-item.ts
@@ -25,6 +25,6 @@ export class BundleItem {
     }
 
     public isTypescriptFile(): boolean {
-        return this.filename && !this.isTypingsFile() && /\.(ts|tsx)$/.test(this.filename);
+        return this.filename && !this.isTypingsFile() && /\.(ts|tsx|json)$/.test(this.filename);
     }
 }

--- a/tests/integration-latest/src/imports/json-import/json-direct-import.ts
+++ b/tests/integration-latest/src/imports/json-import/json-direct-import.ts
@@ -1,0 +1,7 @@
+
+import pkgData from "stream-http/package.json";
+import data from "./json-import-tester.json";
+
+
+export const local = data;
+export const pkg = pkgData;

--- a/tests/integration-latest/src/imports/json-import/json-import-tester.spec.ts
+++ b/tests/integration-latest/src/imports/json-import/json-import-tester.spec.ts
@@ -18,4 +18,10 @@ describe("JsonImporter", () => {
 
         expect(tester.testRequireModuleRequiringJSON()).not.toBeUndefined();
     });
+
+    it("should support TS import of a json file", () => {
+        const { local, pkg } = tester.testTSImportJSON();
+        expect(local).toEqual([1, 2, 3, "a", "b", "c"]);
+        expect(pkg).not.toBeUndefined();
+    });
 });

--- a/tests/integration-latest/src/imports/json-import/json-import-tester.ts
+++ b/tests/integration-latest/src/imports/json-import/json-import-tester.ts
@@ -15,4 +15,9 @@ export class JsonImportTester {
         // globals@9.14.0 requires a JSON file
         return require("globals");
     }
+
+    public testTSImportJSON(): any {
+
+        return require("./json-direct-import");
+    }
 }

--- a/tests/integration-latest/tsconfig.json
+++ b/tests/integration-latest/tsconfig.json
@@ -8,6 +8,7 @@
         "module": "commonjs",
         "noImplicitAny": true,
         "outDir": "tmp",
+        "resolveJsonModule": true,
         "target": "ES5",
         "allowSyntheticDefaultImports": true,
         "baseUrl": ".",

